### DISCUSSION TITLE
How can I retrieve extension information (Authors, Version)?

### DISCUSSION THREAD

**Discussion by coatless** [quarto-dev/quarto-cli#8559] [Q&A] [answered]
### Description

I'm trying to create an extension version-stamped document. Think of it similar to appending `sessionInfo()` at the end of an Rmd document. 

In R, we can grab the version of a package with: 

```md
Version: '`r packageVersion("mypkg")`'
```

Is there a way to access information for a given Quarto extension that is loaded in the document in a similar vein? e.g.

```md
Version: {{< qext version 'my-extension'>}}
```

**Comment by mcanouil** [comment, marked as answer]
There is no built-in way.
Note that by default when you use extensions, those are installed from the default branch, so independently of the version/release.

Since the structure of an extension is standardise, a simple Lua filter/shortcode that reads the YAML for each extensions the `_extension` directory should do the job.

**Reply by coatless** [reply]
Awesome, thanks! I thought as much on this one.

**Reply by mcanouil** [reply]
By the way, if you write something to do this. Writing only the version won't help much as from that you will still need to figure out the GitHub repository (or local archive).

For example:

```sh
quarto add mcanouil/quarto-iconify@v1.0.8
```

will create: `_extensions/mcanouil/iconify`, even when retrieving the version in `_extension.yml`, you cannot find back when it comes from.

In some aspects, the version is not that useful.

**Reply by coatless** [reply]
@mcanouil It's helpful to know what extensions are on the loaded page. For this particular case, I was trying to document the version of the extension that was used to generate the page in an automated way. 

Plus, I'm okay with detecting `user/extension` or just `extension`; no need for knowing/detecting true providence as a quick google/github search should yield the outcomes. 

That said, maybe this could be added to the extension spec in Quarto 1.5?

**Comment by cscheid** [comment]
I do wonder if we should expose this information in Lua.  @coatless What's your use case here? I'm just trying to collect additional information. Thanks!

**Reply by coatless** [reply]
@cscheid main use case is being able to specify the extension version information associated with documentation that is being written and making sure that reprex is able to retrieve version information.

Eventually, I would love to be able to create a `sessionInfo()` component for rendered Quarto and include it at the end of the documents. 

So something like: 
```
#> ─ Session info ───────────────────────────────────────────────────
#>  setting  value
#>  os       Ubuntu 22.04.3 LTS
#>  tz       UTC
#>  date     2024-02-03
#>  quarto   1.4.549
#> 
#> ─ Packages ───────────────────────────────────────────────────────
#>  extension    version date (UTC) source
#>  ext1         3.3.1   2024-01-15 user/repo
#>  ext2         4.0.5   2023-12-30 user/repo2
#>  ext3         0.3.4   2023-12-10 user/repo3
#>  ext4         5.4.8   2023-08-01 user/repo4
```

I think I'll explore the extension parsing in a shortcode extension.

**Reply by coatless** [reply]
I took a stab at implementing the desired logic as a Quarto extension in <https://github.com/coatless-quarto/stamp/>
